content/doc/book/pipeline/shared-libraries.adoc

```diff
@@ -471,7 +466,14 @@ Reference_ (under _Pipeline Syntax_) after Jenkins loads and uses that library
 as part of a successful Pipeline run.
 ====
 
-=== Defining steps
+.Avoid preserving state in global variables
+[WARNING]
+====
+Avoid defining global variables with methods that interact or preserve state.
+Use a static class or instantiate a local variable of a class instead.
+====
+
+=== Defining custom steps
```

Comment:
I would not advise referring to these as “steps” BTW. They are just functions that might look kind of like steps (or might not).